app/controllers/api/v2/providers_controller.rb

```diff
@@ -35,6 +35,8 @@ def update
         update_ucas_preferences
 
         if @provider.valid?
+          courses_synced?(@provider.syncable_courses) if @recruitment_cycle.current? && @provider.syncable_courses.present?
```

Comment:
Interestingly, we do a `@provider.reload` next line down ... which means that if anything has changed the provider record in the calls to `update_*` above (`update_enrichment`, etc), then the provider record will be updated. I think this is significant because if we expect that provider needs to be reloaded, then `@provider.syncable_courses` might be working off of stale data.

I think we should ensure we do `@provider.reload` before we call `@provider.syncable_courses`, and probably even before we do `@provider.valid?`.

Also, imo this line is too long to have everything on one line like this. Maybe turn it into a multi-line if-statement?

Reply:
As per IRL discussion, the `@provider.reload` below is probably miss-placed and should really just be removed. Resolving for now.